### DISCUSSION TITLE
Add foonathan as a third-party [13817]

### DISCUSSION THREAD

**Discussion by Nir-Az** [eProsima/Fast-DDS#2418] [Ideas]
Hi,

I am integrating FastDDS into a Cmake project and I was wondering, `foonathan_memory` is a hard dependency of FastDDS, same as `asio` and `tinyxml`.
Why is it threated differently inside the cmake? wasn't it better if it could have been integrated automatically with
 `-DTHIRDPARTY_foonathan_memory=FORCE`?

Instead the user have to make sure he integrate it before adding FastDDS project.
And make sure the versions match and the API is not broken..

Thanks
Nir

**Comment by xyzhai-pro** [comment]
hello, I am integrating FastDDS into a Cmake project by using Xxx-config.cmake in "cmake" folder. However, i failed. Could you tell me how to integrate FastDDS into Cmake project.

**Reply by Nir-Az** [reply]
Hi @xyzhai-pro ,

As I know how frustrating it is to make this complicated integration (and got no response here about it ) I will share how I made it happen (Works on Linux + Windows) 

I am sure there are several ways to do it.

[Main Cmake addition](https://github.com/IntelRealSense/librealsense/blob/42e94f8d2b489161568f59109e6c38362683c5ed/CMakeLists.txt#L70)
[external_foonathan_memory.cmake](https://github.com/IntelRealSense/librealsense/blob/dds/CMake/external_foonathan_memory.cmake)
[external_fastdds.cmake](https://github.com/IntelRealSense/librealsense/blob/dds/CMake/external_fastdds.cmake)

Too bad the integration of hard dependencies (Like `Foonathan-memory`) is not taking care off as part of `Fast-DDS`, it will surly help the users..
Good luck.
